handlers/geolocation/geolocation.go

```diff
@@ -257,6 +297,11 @@ func (c *GeolocationHandlersCollection) getStreamPull(playbackID string, retryCo
 		return "", nil
 	}
 
+	// To prevent overloading the LAPI, we limit the rate at which we query for stream pulls
+	if c.streamPullRateLimit.shouldLimit(playbackID) {
```

Comment:
Why did you need to change it from just a single `acquire` call? Doesn't this mean that there will be a window of time between the call to `shouldLimit` and `acquire` where multiple routines could call `acquire`?

Reply:
Yes, but it's ok for multiple goroutines to `acquire()` in parallel. But your comment made me think I should rename this function to `mark()`. Otherwise, it sounds like a semaphore.

And to your question, I changed it because I realized that this function is also executed for streams that are not pull, but just ingest. Then, we don't want to make any retries or rate limit for those. In particular, if there are multiple playback requests at the same to the same non-active stream, we don't want to retry it.